delfin/drivers/dell_emc/vmax/client.py

```diff
@@ -38,52 +35,48 @@ def __init__(self, **kwargs):
         self.rest_port = rest_access.get('port')
         self.rest_username = rest_access.get('username')
         self.rest_password = rest_access.get('password')
-
-    def __del__(self):
-        # De-initialize session
-        if self.conn:
-            self.conn.close_session()
-            self.conn = None
+        self.rest = rest.VMaxRest()
+        self.rest.set_rest_credentials(rest_access,
+                                       kwargs.get('extra_attributes', None))
 
     def init_connection(self, access_info):
         """ Given the access_info get a connection to VMAX storage """
-        self.array_id = access_info.get('extra_attributes', {}). \
-            get('array_id', None)
-        if not self.array_id:
-            raise exception.InvalidInput('Input array_id is missing')
 
         try:
-            # Initialise PyU4V connection to VMAX
-            self.conn = PyU4V.U4VConn(
-                u4v_version=SUPPORTED_VERSION,
-                server_ip=self.rest_host,
-                port=self.rest_port,
-                verify=False,
-                array_id=self.array_id,
-                username=self.rest_username,
-                password=self.rest_password)
-
+            ver, self.uni_version = self.rest.get_uni_version()
```

Comment:
Do we need to validate for a minmum version of unisphere? what all are supported versions with this client ?

Reply:
It should support older version, minimum version tested can be documented as 90.